MantidPlot/src/ProjectRecovery.cpp

```diff
@@ -435,15 +437,16 @@ void ProjectRecovery::saveWsHistories(const Poco::Path &historyDestFolder) {
   alg->setLogging(false);
 
   for (const auto &ws : wsHandles) {
-    std::string filename = ws;
+    std::string filename = ws->getName();
```

Comment:
Can we avoid copy the string and use `const auto &`?

Reply:
Another thought just occurred. What about the case where a workspace name has characters that are invalid in a file path? https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations

Reply:
It was reported this morning: #22780 
I'll look at addressing it in a future PR

Reply:
Unfortunately we can't avoid this copy, since we need to append '.py' to the filename

Reply:
:+1: